test/test_transforms.py

```diff
@@ -796,25 +806,25 @@ def test_pil_to_tensor(self):
             input_data = torch.ByteTensor(channels, height, width).random_(0, 255)
             img = transforms.ToPILImage()(input_data)
             output = trans(img)
-            self.assertTrue(np.allclose(input_data.numpy(), output.numpy()))
+            torch.testing.assert_close(input_data, output, check_stride=False)
 
             input_data = np.random.randint(low=0, high=255, size=(height, width, channels)).astype(np.uint8)
             img = transforms.ToPILImage()(input_data)
             output = trans(img)
             expected_output = input_data.transpose((2, 0, 1))
-            self.assertTrue(np.allclose(output.numpy(), expected_output))
+            torch.testing.assert_close(output.numpy(), expected_output)
 
             input_data = torch.as_tensor(np.random.rand(channels, height, width).astype(np.float32))
             img = transforms.ToPILImage()(input_data)  # CHW -> HWC and (* 255).byte()
             output = trans(img)  # HWC -> CHW
             expected_output = (input_data * 255).byte()
-            self.assertTrue(np.allclose(output.numpy(), expected_output.numpy()))
+            torch.testing.assert_close(output, expected_output, check_stride=False)
 
         # separate test for mode '1' PIL images
         input_data = torch.ByteTensor(1, height, width).bernoulli_()
         img = transforms.ToPILImage()(input_data.mul(255)).convert('1')
-        output = trans(img)
-        self.assertTrue(np.allclose(input_data.numpy(), output.numpy()))
+        output = trans(img).view(torch.uint8).bool().to(torch.uint8)
```

Comment:
This looks really weird but is actually needed. Although the image is binary (`.convert("1")`), the data is stored as byte per pixel instead of one bit. Although `trans(img).dtype == torch.bool` in memory it is still stored as `255`. Naively doing `trans(img).to(torch.uint8)` would result in values of `0` and `255`.

Thus, we look at the values as if they are `uint8` (since they technically are) (`.view(torch.uint8)`), convert this to `bool` to write the proper values into the memory (`.bool()`) and finally convert it back to `uint8` (`.to(torch.uint8)`).

This never came up before, because `np.allclose` internally always converts both inputs to `np.float64`, whereas `torch.testing.assert_close` only converts to the result of [`torch.promote_type`](https://pytorch.org/docs/stable/generated/torch.promote_types.html?highlight=promote_type), which gives `torch.uint8` for inputs of `torch.bool` and `torch.uint8`.